plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring.go

```diff
@@ -0,0 +1,244 @@
+func (a *NebiusCloudMonitoring) Close() error {
+	a.client = nil
+	return nil
+}
+
+// Write writes metrics to the remote endpoint
+func (a *NebiusCloudMonitoring) Write(metrics []telegraf.Metric) error {
+	var nebiusCloudMonitoringMetrics []nebiusCloudMonitoringMetric
+	for _, m := range metrics {
+		for _, field := range m.FieldList() {
+			value, err := internal.ToFloat64(field.Value)
+			if err != nil {
+				a.Log.Errorf("skipping value: %w", err.Error())
+				continue
+			}
+
+			nebiusCloudMonitoringMetrics = append(
+				nebiusCloudMonitoringMetrics,
+				nebiusCloudMonitoringMetric{
+					Name:   m.Name() + "_" + field.Key,
+					Labels: m.Tags(),
+					TS:     m.Time().Format(time.RFC3339),
+					Value:  value,
+				},
+			)
+		}
+	}
+
+	body, err := json.Marshal(
+		nebiusCloudMonitoringMessage{
+			Metrics: nebiusCloudMonitoringMetrics,
+		},
+	)
+	if err != nil {
+		return err
+	}
+	body = append(body, '\n')
+	return a.send(body)
+}
+
+func (a *NebiusCloudMonitoring) getResponseFromMetadata(c *http.Client, metadataURL string) ([]byte, error) {
+	req, err := http.NewRequest("GET", metadataURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+	req.Header.Set("Metadata-Flavor", "Google")
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode >= 300 || resp.StatusCode < 200 {
+		return nil, fmt.Errorf("unable to fetch instance metadata: [%s] %d",
+			metadataURL, resp.StatusCode)
+	}
+	return body, nil
+}
+
+func (a *NebiusCloudMonitoring) getIAMTokenFromMetadata() (string, int, error) {
+	a.Log.Debugf("getting new IAM token in %s", a.metadataTokenURL)
+	body, err := a.getResponseFromMetadata(a.client, a.metadataTokenURL)
+	if err != nil {
+		return "", 0, err
+	}
+	var metadata metadataIamToken
+	if err := json.Unmarshal(body, &metadata); err != nil {
+		return "", 0, err
+	}
+	if metadata.AccessToken == "" || metadata.ExpiresIn == 0 {
+		return "", 0, fmt.Errorf("unable to fetch authentication credentials %s: %w", a.metadataTokenURL, err)
+	}
+	return metadata.AccessToken, int(metadata.ExpiresIn), nil
+}
+
+func (a *NebiusCloudMonitoring) send(body []byte) error {
+	req, err := http.NewRequest("POST", a.Endpoint, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+	q := req.URL.Query()
+	q.Add("folderId", a.folderID)
+	q.Add("service", a.Service)
+	req.URL.RawQuery = q.Encode()
+
+	req.Header.Set("Content-Type", "application/json")
+	isTokenExpired := a.iamTokenExpirationTime.Before(time.Now())
+	if a.iamToken == "" || isTokenExpired {
+		token, expiresIn, err := a.getIAMTokenFromMetadata()
+		if err != nil {
+			return err
+		}
+		a.iamTokenExpirationTime = time.Now().Add(time.Duration(expiresIn) * time.Second)
+		a.iamToken = token
+	}
+	req.Header.Set("Authorization", "Bearer "+a.iamToken)
+
+	a.Log.Debugf("sending metrics to %s", req.URL.String())
```

Comment:
do we need this printed at every send? I don't mind having the endpoint printed during plugin startup